pkg/proc/breakpoints.go

```diff
@@ -609,11 +620,22 @@ func (t *Target) setBreakpointInternal(addr uint64, kind BreakpointKind, wtype W
 	newBreaklet := &Breaklet{Kind: kind, Cond: cond}
 	if kind == UserBreakpoint {
 		newBreaklet.HitCount = map[int]uint64{}
+		bpmap.breakpointIDCounter++
+		newBreaklet.LogicalID = bpmap.breakpointIDCounter
 	}
 	if bp, ok := bpmap.M[addr]; ok {
 		if !bp.canOverlap(kind) {
 			return bp, BreakpointExistsError{bp.File, bp.Line, bp.Addr}
 		}
+		if kind == UserBreakpoint {
+			bp.Tracepoint = false
```

Comment:
Why set all these here?

Reply:
They are only relevant for user breakpoints if they are set it means they had been set for an old user breakpoint that was removed and kept around by overlapping non-user breakpoints.